src/main/java/fr/insee/pogues/transforms/visualize/composition/UpdateIterationBounds.java

```diff
@@ -0,0 +1,70 @@
+package fr.insee.pogues.transforms.visualize.composition;
+
+import fr.insee.pogues.exception.DeReferencingException;
+import fr.insee.pogues.exception.IllegalIterationException;
+import fr.insee.pogues.model.ComponentType;
+import fr.insee.pogues.model.IterationType;
+import fr.insee.pogues.model.Questionnaire;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+
+import static fr.insee.pogues.utils.PoguesModelUtils.getIterationBounds;
+import static fr.insee.pogues.utils.PoguesModelUtils.getSequences;
+
+/**
+ * Implementation of CompositionStep to update Iteration (loops) objects when de-referencing a questionnaire.
+ */
+@Slf4j
+class UpdateIterationBounds implements CompositionStep {
+
+    /**
+     * Update iterations of the referencing questionnaire: if a start/end member of an iteration is a referenced
+     * questionnaire, replace the reference id by the right element's id from the referenced questionnaire.
+     * @param questionnaire Referencing questionnaire.
+     * @param referencedQuestionnaire Referenced questionnaire.
+     * @throws DeReferencingException if an error occurs during iterations update.
+     */
+    @Override
+    public void apply(Questionnaire questionnaire, Questionnaire referencedQuestionnaire)
+            throws DeReferencingException {
+        if (questionnaire.getIterations() != null) {
+            try {
+                for (IterationType iterationType : questionnaire.getIterations().getIteration()) {
+                    updateIterationBounds(referencedQuestionnaire, iterationType);
+                }
+                log.info("Iterations' bounds updated in '{}' when de-referencing '{}'",
+                        questionnaire.getId(), referencedQuestionnaire.getId());
+            } catch (IllegalIterationException e) {
+                String message = String.format(
+                        "Error when updating iteration bounds in questionnaire '%s' with reference '%s'",
+                        questionnaire.getId(), referencedQuestionnaire.getId());
+                throw new DeReferencingException(message, e);
+            }
+        }
+    }
+
+    /** Replace loop bounds that reference a questionnaire by its first or last sequence.
+     * @param referencedQuestionnaire Referenced questionnaire.
+     * @param iterationType The Iteration object to be updated.
+     * @throws IllegalIterationException if the 'MemberReference' property in the iteration is invalid.
+     */
+    static void updateIterationBounds(Questionnaire referencedQuestionnaire, IterationType iterationType)
+            throws IllegalIterationException {
+        //
+        String reference = referencedQuestionnaire.getId();
+        //
+        List<String> iterationBounds = getIterationBounds(iterationType);
+        // Replace questionnaire reference by its first/last sequence
+        String beginMember = iterationBounds.get(0);
```

Comment:
get(0), get(1), ... are we sure it's always defined (i think i remembered there was a check earlier, but not sure).

Reply:
Yes, the `getIterationBounds` method is designed for this and always returns a list of 2 elements (as it docs says). The design is not so good yet, but the cause of this is a flawed modeling of this property in current's Pogues-Model...